### ISSUE TITLE
On Guava 22: NoSuchMethodError: com.google.common.net.HostAndPort.fromHost

### ISSUE THREAD

**Issue by seanf** [lightbody/browsermob-proxy#638]
This is like https://github.com/lightbody/browsermob-proxy/issues/398, except this time the method `getHostText` is missing because it is being _removed_ from Guava. It's considered fair game because the class `com.google.common.net.HostAndPort` is marked `@Beta`. See https://github.com/google/guava/wiki/PhilosophyExplained#beta-apis

This could be handled by switching from `getHostText` to `getHost`, but this would force users to upgrade to Guava ~22~ 20 (and upgrading Guava is often difficult, precisely because of problems like this). 

I would recommend avoiding Guava's `@Beta` classes/methods altogether (or perhaps Guava generally) in any library code, because they eventually lead to maintenance problems. Another option is to repackage the guava jar.

FindBugs with the extra detector GuavaBetaDetector can help to prevent `@Beta` API calls from creeping in again:

```xml
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>findbugs-maven-plugin</artifactId>
                <configuration>
                    <plugins>
                        <plugin>
                            <groupId>com.overstock.findbugs</groupId>
                            <artifactId>library-detectors</artifactId>
                            <version>1.1</version>
                        </plugin>
                    </plugins>
                </configuration>
                <executions>
                    <execution>
                        <id>default</id>
                        <phase>test</phase>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

**Comment by seanf** [comment]
Correction: switching from `getHostText` to `getHost` would force users to upgrade to Guava 20+, not Guava 22.

**Comment by jekh** [comment]
Thanks for noting this. The issue has been resolved by #666.

It's unfortunate that guava still considers HostAndPort to be "beta". It's been in guava since 10.0 (Sept 2011). It's generally quite useful and I'd rather not use a roll-your-own solution, but it's unfortunate that they are so quick to remove functionality just because they technically haven't removed the "beta" designation after 6 years.
